src/pixelator/graph/backends/implementations.py

```diff
@@ -628,157 +697,335 @@ def add_names_to_vertexes(self, vs_names: List[str]) -> None:
         raise NotImplementedError()
 
 
+class IgraphBasedVertex(Vertex):
+    """A Vertex instance backed by an igraph.Vertex."""
+
+    def __init__(self, vertex: igraph.Vertex):
+        """Create a new IgraphBasedVertex instance."""
+        if not isinstance(vertex, igraph.Vertex):
+            raise TypeError(f"{vertex} is not `igraph.Vertex`")
+        self._vertex = vertex
+
+    @property
+    def index(self):
+        """Get the index of the vertex."""
+        return self._vertex.index
+
+    def __getitem__(self, attr: str) -> Any:
+        """Get the attr of the provided vertex."""
+        return self._vertex[attr]
+
+    def __setitem__(self, attr: str, value: Any) -> None:
+        """Get the attr of the provided vertex."""
+        self._vertex[attr] = value
+
+    def neighbors(self):
+        """Fetch all the neighboring vertices."""
+        return self._vertex.neighbors()
+
+
+class IgraphBasedEdge(Edge):
+    """An Edge instance backed by an igraph.Edge."""
+
+    def __init__(self, edge: igraph.Edge):
+        """Create an IgraphBasedEdge instance."""
+        self._edge = edge
+
+    @property
+    def index(self) -> int:
+        """The index of the edge."""
+        return self._edge.index
+
+    @property
+    def vertex_tuple(self) -> Tuple[Vertex, Vertex]:
+        """Return the vertices the edge connects as a tuple."""
+        v1, v2 = self._edge.vertex_tuple
+        return (IgraphBasedVertex(v1), IgraphBasedVertex(v2))
+
+
 class IgraphBasedVertexSequence(VertexSequence):
     """Proxy for a igraph.VertexSeq."""
 
-    def __init__(self, raw) -> None:
+    def __init__(self, vertex_sequence: igraph.VertexSeq, graph: igraph.Graph) -> None:
         """Instantiate a new IgraphBasedVertexSequence."""
-        self._raw = raw
+        self._vertex_seq = vertex_sequence
+        self._graph = graph
+
+    def vertices(self) -> Iterable[Vertex]:
+        """Return an iterable of vertices."""
+        return [IgraphBasedVertex(vertex) for vertex in self._vertex_seq]
 
     def select(self, **kwargs):
         """Select a subset of vertices.
 
         See https://python.igraph.org/en/stable/api/igraph.VertexSeq.html#select
         """
-        return self._raw.select(**kwargs)
+        # TODO This needs to be fixed to actually wrap a vertex and be
```

Comment:
Just a reminder. Do we have a ticket for this?

Reply:
I'll go ahead and fix it now, since it's actually only used in the tests anyway.